include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -173,14 +180,33 @@ struct cartesian_segments
         return disjoint_box_box_strategy_type();
     }
 
-    typedef disjoint::segment_box disjoint_segment_box_strategy_type;
+    typedef disjoint::segment_box
+    <
+        CalculationType,
+        distance::pythagoras<CalculationType>
+    > disjoint_segment_box_strategy_type;
```

Comment:
Hmm, I probably missed it, but why does `disjoint::segment_box` require distance strategy?

Reply:
ooops, it doesn't. It was there before this PR but I am removing it now.